src/tests/simulation/tests/sink.rs

```diff
@@ -0,0 +1,216 @@
+    pub after: Option<After>,
+    pub before: Option<Before>,
+    pub op: String,
+    #[serde(rename = "ts_ms")]
+    pub ts_ms: i64,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct After {
+    pub count: i64,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Before {
+    pub count: i64,
+}
+
+#[madsim::test]
+async fn test_sink_append_only() -> Result<()> {
+    let mut cluster = Cluster::start(Configuration::for_scale()).await?;
+
+    let mut topics = HashMap::new();
+    topics.insert("t_sink_append_only".to_string(), 3);
+    cluster.create_kafka_topics(topics);
+
+    time::sleep(Duration::from_secs(10)).await;
+
+    cluster.run(ROOT_TABLE_CREATE).await?;
+    cluster.run(APPEND_ONLY_SINK_CREATE).await?;
+
+    let materialize_fragment = cluster
+        .locate_one_fragment([
+            identity_contains("materialize"),
+            no_identity_contains("globalSimpleAgg"),
+        ])
+        .await?;
+
+    let id = materialize_fragment.id();
+    cluster.reschedule(format!("{id}-[1,2,3,4,5]")).await?;
+
+    let consumer: StreamConsumer<_> = cluster
+        .run_on_client(async move {
+            let consumer = ClientConfig::new()
+                .set("bootstrap.servers", "192.168.11.1:29092")
+                .set("group.id", "id")
+                .create::<StreamConsumer<_>>()
+                .await
+                .expect("failed to create kafka consumer client");
+
+            let mut tpl = TopicPartitionList::new();
+
+            for i in 0..3 {
+                tpl.add_partition(APPEND_ONLY_TOPIC, i);
+            }
+
+            consumer.assign(&tpl).unwrap();
+            consumer
+        })
+        .await;
+
+    let mut stream = consumer.stream();
+
+    for i in &[1, 2, 3, 4, 5] {
+        cluster
+            .run(&format!("insert into t values ({});", i))
+            .await?;
+        cluster.run(&format!("flush;")).await?;
+
+        let msg = stream.next().await.unwrap().unwrap();
+        let payload = msg.payload().unwrap();
+
+        check_payload(&msg, payload, *i);
+    }
+
+    Ok(())
+}
+
+#[madsim::test]
+async fn test_sink_debezium() -> Result<()> {
+    let mut cluster = Cluster::start(Configuration::for_scale()).await?;
+
+    let mut topics = HashMap::new();
+    topics.insert("t_sink_debezium".to_string(), 3);
+    cluster.create_kafka_topics(topics);
+
+    time::sleep(Duration::from_secs(10)).await;
+
+    cluster.run(ROOT_TABLE_CREATE).await?;
+    cluster.run(MV_CREATE).await?;
+    cluster.run(DEBEZIUM_SINK_CREATE).await?;
+
+    let materialize_fragment = cluster
+        .locate_one_fragment([
+            identity_contains("materialize"),
+            no_identity_contains("globalSimpleAgg"),
+        ])
+        .await?;
+
+    let id = materialize_fragment.id();
+    cluster.reschedule(format!("{id}-[1,2,3,4,5]")).await?;
```

Comment:
What about also inserting and checking the sink records before scaling?

Reply:
make sense, i'll add some test